akka-stream/src/main/scala/akka/stream/stage/StageLogging.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.stage
+
+import akka.event.{ LoggingAdapter, NoLogging }
+import akka.stream.{ ActorMaterializer, MaterializerLoggingProvider }
+
+/**
+ * Simple way to obtain a [[LoggingAdapter]] when used together with an [[ActorMaterializer]].
+ * If used with a different materializer [[NoLogging]] will be returned.
```

Comment:
does this work with genjavadoc, or do we have to qualify the links with full package name?